### ISSUE TITLE
Temporary data structures

### ISSUE THREAD

**Issue by Brunix90** [fracpete/python-weka-wrapper#65]
Hello,

I am using the library to perform some cross-validation tuning on Python, on macOS 10.14, through Jupyter Notebook.

On each fold iteration, before calling Weka, I start the JVM through jvm.start(), and then I stop it through jvm.stop().

However, I am noticing an ever increasing slowdown as tuning iterations go on.
Moreover, the disk spaces shrinks: at the beginning I have 24  GB free, then it reduces to 18, 15, .... all the way down to 3.

At that time the notebook becomes really underperforming, and I have to reboot the system.
After the reboot, the situation is normal again: 24 GB free, and decent performance.

It seems like even stopping the JVM doesn't allow the system to free some temporary structures, that become larger and larger as the tuning iterations go on.

**Comment by fracpete** [comment]
Past versions of javabridge (which pww uses under the) had the limitation of not being able to restart the jvm again in the same process once stopped. My guess is, that this is still not working reliably and the jupyter kernel is holding onto the JVM still. The jvm should only be started once and I recommend not using jupyter, but just regular Python scripts, eg using PyCharm or another IDE.

**Comment by Brunix90** [comment]
Thank you for the fast reply, I will try your suggestion!

-A.

**Comment by fracpete** [comment]
I never encountered that problem before. Are you using an Oracle JDK?

**Comment by Brunix90** [comment]
This is the version I am using:

java version "1.8.0_60"
Java(TM) SE Runtime Environment (build 1.8.0_60-b27)
Java HotSpot(TM) 64-Bit Server VM (build 25.60-b23, mixed mode)

It is strange indeed, maybe also this is connected to Jupyter Notebook.

Basically, I execute jvm.start() and train a model in a step, and then do some other things not involving the wrapper for some minutes (in subsequent cells). Finally, when I try to call again the model I trained initially, that results into an error.
If instead I trained the model, pickled it to disk, closed the vm, and then at a later time reopened the JVM, reloaded the model and executed it, all works just fine.

I will keep you posted.
